HARK/ConsumptionSaving/ConsIndShockModel.py

```diff
@@ -2659,77 +2660,72 @@ def construct_lognormal_income_process_unemployment(self):
+                                                       'IncUnemp': IncUnemp_list,
+                                                       'n_approx': TranShkCount_list})
+
+        IncShkDstn = IndexDistribution(engine = InkShk_engine,
+                                       conditional = {'sigma_Perm': PermShkStd,
+                                                      'sigma_Tran': TranShkStd,
+                                                      'n_approx_Perm': PermShkCount_list,
+                                                      'n_approx_Tran': TranShkCount_list,
+                                                      'UnempPrb': UnempPrb_list,
+                                                      'IncUnemp': IncUnemp_list},
+                                       seed=self.RNG.randint(0, 2 ** 31 - 1))
 
-        # Loop to fill in the list of IncShkDstn random variables.
-        for t in range(T_cycle):  # Iterate over all periods, counting forward
+        return IncShkDstn, PermShkDstn, TranShkDstn
 
-            if T_retire > 0 and t >= T_retire:
-                # Then we are in the "retirement period" and add a retirement income object.
-                IncShkDstn.append(deepcopy(IncShkDstnRet))
-                PermShkDstn.append([np.array([1.0]), np.array([1.0])])
-                TranShkDstn.append([ShkPrbsRet, TranShkValsRet])
-            else:
-                # We are in the "working life" periods.
-                TranShkDstn_t = MeanOneLogNormal(sigma=TranShkStd[t]).approx(
-                    TranShkCount, tail_N=0
-                )
-                if UnempPrb > 0:
-                    TranShkDstn_t = add_discrete_outcome_constant_mean(
-                        TranShkDstn_t, p=UnempPrb, x=IncUnemp
-                    )
-                PermShkDstn_t = MeanOneLogNormal(sigma=PermShkStd[t]).approx(
-                    PermShkCount, tail_N=0
-                )
+def PermShk_engine(sigma, n_approx, seed = 0):
 
-                IncShkDstn.append(
-                    combine_indep_dstns(
-                        PermShkDstn_t,
-                        TranShkDstn_t,
-                        seed=self.RNG.randint(0, 2 ** 31 - 1),
-                    )
-                )  # mix the independent distributions
-                PermShkDstn.append(PermShkDstn_t)
-                TranShkDstn.append(TranShkDstn_t)
-        return IncShkDstn, PermShkDstn, TranShkDstn
+    PermShkDstn = MeanOneLogNormal(sigma, seed = seed).approx(n_approx, tail_N=0)
+
+    return PermShkDstn
+
+def TranShk_engine(sigma, UnempPrb, IncUnemp, n_approx, seed = 0):
+
+    TranShkDstn = MeanOneLogNormal(sigma, seed = seed).approx(n_approx, tail_N=0)
+    if UnempPrb > 0:
+        TranShkDstn = add_discrete_outcome_constant_mean(TranShkDstn, p=UnempPrb, x=IncUnemp)
+
+    return TranShkDstn
+
+def InkShk_engine(sigma_Perm, sigma_Tran, n_approx_Perm, n_approx_Tran, UnempPrb, IncUnemp, seed = 0):
```

Comment:
I believe this should be IncShk ... with a 'c'

Reply:
True, thank you!